upup/pkg/fi/cloudup/awstasks/iamrolepolicy.go

```diff
@@ -43,12 +43,49 @@ type IAMRolePolicy struct {
 
 	// The PolicyDocument to create as an inline policy.
 	// If the PolicyDocument is empty, the policy will be removed.
-	PolicyDocument fi.Resource
+	PolicyDocument  fi.Resource
+	PolicyOverrides *[]string
+	Managed         bool
 }
 
 func (e *IAMRolePolicy) Find(c *fi.Context) (*IAMRolePolicy, error) {
+	var actual IAMRolePolicy
+
 	cloud := c.Cloud.(awsup.AWSCloud)
 
+	// Handle policy overrides
+	if e.PolicyOverrides != nil {
```

Comment:
Maybe this should be a separate task (e.g. IAMRolePolicyAttachements) - or we should raise an error if we specify PolicyOverrides and PolicyDocument on the same task.

Creating a new task is a trickier thing normally, so I'm happy to do that separately...

Reply:
I actually did try! I couldn't quite get it figured out.